parcels/particleset.py

```diff
@@ -516,11 +516,11 @@ def density(self, field=None, particle_val=None, relative=False, area_scale=Fals
         density = np.zeros((field.grid.lat.size, field.grid.lon.size), dtype=np.float32)
 
         for pi, p in enumerate(self.particles):
-            try:  # breaks if either p.gridIndexSet does not exist (in scipy) or field not in fieldset
-                if p.gridIndexSet[field.grid].ti < 0:  # xi, yi, ti, not initialised
+            try:  # breaks if either p.xis do not exist (in scipy) or field not in fieldset
+                if p.tis[field.iGrid] < 0:  # xis, yis, tis, not initialised
```

Comment:
So the comment says it's checking for `p.xis` but the actual check is on `p.tis`?